packages/std-client/src/utils.ts

```diff
@@ -117,16 +100,61 @@ export function resolveRelationshipChain(
   return entity;
 }
 
+export function findComponentInRelationshipChain(
+  entity: EntityIndex,
+  searchComponent: AnyComponent,
+  world: World,
+  relationshipComponent: Component<{ value: Type.Entity }, { contractId: string }>
+): EntityIndex | undefined {
+  if (hasComponent(searchComponent, entity)) return entity;
+
+  while (hasComponent(relationshipComponent, entity)) {
+    const entityValue = world.entityToIndex.get(getComponentValueStrict(relationshipComponent, entity).value);
+    if (!entityValue) return;
+    entity = entityValue;
+
+    if (hasComponent(searchComponent, entity)) return entity;
+  }
+
+  return;
+}
+
+export function findInRelationshipChain(
+  entity: EntityIndex,
+  searchEntity: EntityIndex,
+  world: World,
+  relationshipComponent: Component<{ value: Type.Entity }, { contractId: string }>
+): EntityIndex | undefined {
+  if (entity === searchEntity) return entity;
+
+  while (hasComponent(relationshipComponent, entity)) {
+    const entityValue = world.entityToIndex.get(getComponentValueStrict(relationshipComponent, entity).value);
+    if (!entityValue) return;
```

Comment:
same as above, `entityValue` could be 0, so we should do a null check instead
```suggestion
    if (entityValue == null) return;
```